lib/Common/CommonPal.h

```diff
@@ -4,6 +4,12 @@
 //-------------------------------------------------------------------------------------------------------
 #pragma once
 
+// Define _ALWAYSINLINE for template that want to always inline, but doesn't allow inline linkage in clang
+#if defined(__GNUC__) || defined(__clang__)
+#define _ALWAYSINLINE __attribute__((always_inline))
```

Comment:
Why don't we `#define __forceinline  __attribute__((always_inline))` and then use `__forceinline` cross-platform?